### ISSUE TITLE
Unable to Dispose() 

### ISSUE THREAD

**Issue by Floyddotnet** [voloda/AsyncDataAdapter#2]
**Current Behavior**

System.InvalidCastException: Unable to cast object of type 'AsyncDataAdapter.SqlDataAdapter' to type 'System.Data.IDbDataAdapter'

**Link to Source**

[at AsyncDataAdapter.DbDataAdapter.Dispose(Boolean disposing)](https://github.com/voloda/AsyncDataAdapter/blob/200e121b0d5b52f85808f3810e99a956ddba96af/AsyncDataAdapter/DbDataAdapter.cs#L294)

**Steps to Reproduce**

```
using (var sqlDa = new AsyncDataAdapter.SqlDataAdapter(sqlCmd))
using (var ds = new DataSet())
{
    await sqlDa.FillAsync(ds);
    return ds;
}
```

**Comment by voloda** [comment]
This should be fixed by PR#3

**Comment by Floyddotnet** [comment]
:thumbsup: thanks
